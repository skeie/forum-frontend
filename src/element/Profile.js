import React, { Component } from 'react';
import { View, Image } from 'react-native';
import { Text } from '../common';

export const Username = ({ username }) => (
    <Text
        style={{
            color: '#55555C',
            marginTop: 20,
            fontSize: 20,
        }}>
        {username}
    </Text>
);

export const ProfileImage = ({ userImage }) => (
    <Image
        source={{ uri: userImage }}
        style={{
            width: 80,
            height: 80,
            borderRadius: 10,
            marginRight: 10,
        }}
    />
);

export const Date = ({ date }) => (
    <Text type="date" numberOfLines={1}>
        {date}
    </Text>
);
