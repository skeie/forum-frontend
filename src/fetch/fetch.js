import queryString from 'query-string';
const baseURl = __DEV__
    ? 'http://192.168.178.190:3000'
    : 'https://fast-spire-38104.herokuapp.com';

let jwtToken = '';

function _appUrl(url) {
    return baseURl + url;
}

export const setJwtToken = token => (jwtToken = token);

export async function getExt(url) {
    try {
        const response = await fetch(url, setHeaders('GET'));
        if (response.status === 200) {
            return Promise.resolve();
        } else {
            return Promise.reject('Next page does not exists');
        }
    } catch (e) {
        return Promise.reject(e);
    }
}

export async function get(url, obj = {}) {
    let constructedUrl = url;
    if (obj.params) {
        constructedUrl += `?${queryString.stringify(obj.params)}`;
    }
    const response = await fetch(_appUrl(constructedUrl), setHeaders('GET'));
    return handleResponse(response);
}

export async function post(url, obj) {
    const response = await fetch(
        _appUrl(url),
        setHeaders('POST', JSON.stringify(obj)),
    );
    return handleResponse(response);
}

export async function postMultiPart(url, obj) {
    const data = new FormData();

    for (const key in obj) {
        data.append(key, JSON.stringify(obj[key]));
    }

    const response = await fetch(
        _appUrl(url),
        setHeaders('POST', data, { 'Content-Type': 'multipart/form-data' }),
    );
    return handleResponse(response);
}

function setHeaders(method, body, optHeader) {
    const headers = Object.assign(
        {
            Accept: 'application/json',
            'Content-Type': 'application/json',
        },
        optHeader,
    );

    if (jwtToken) {
        headers.authorization = `Bearer ${jwtToken}`;
    }

    return {
        method,
        headers,
        body,
    };
}

async function handleResponse(response) {
    if (response.status === 201 || response.status === 204) return;
    let responseJson;
    try {
        responseJson = await response.json();
    } catch (e) {
        throw {
            message: 'Something went wrong, check if you have internet',
        };
    }

    if (response.status >= 400) {
        // check msg or message here and check for json
        const msg =
            responseJson.message ||
            'Something went wrong, check if you have internet';
        throw {
            message: msg,
        };
    } else {
        return responseJson;
    }
}
