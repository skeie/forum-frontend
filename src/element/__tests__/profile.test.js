import React from 'react';
import { Username, ProfileImage, Date } from '../Profile';
import backendData from '../../testData/posts';
import renderer from 'react-test-renderer';
import { post } from '../../fetch/fetch';

it('Username renders correctly', () => {
    const tree = renderer.create(<Username username="dahBest" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('ProfileImage renders correctly', () => {
    const tree = renderer
        .create(<ProfileImage userImage={backendData.posts[0].userImage} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Date renders correctly', () => {
    const tree = renderer.create(<Date date="19/04" />).toJSON();
    expect(tree).toMatchSnapshot();
});
