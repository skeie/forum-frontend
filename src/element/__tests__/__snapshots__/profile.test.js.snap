// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "white",
        "fontFamily": "poppins",
      },
      Object {
        "color": "#DCDDE3",
        "fontFamily": "italic",
      },
      Object {},
    ]
  }
  type="date"
>
  19/04
</Text>
`;

exports[`ProfileImage renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": undefined,
    }
  }
  style={
    Object {
      "borderRadius": 10,
      "height": 80,
      "marginRight": 10,
      "width": 80,
    }
  }
/>
`;

exports[`Username renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "white",
        "fontFamily": "poppins",
      },
      null,
      Object {
        "color": "#55555C",
        "fontSize": 20,
        "marginTop": 20,
      },
    ]
  }
>
  dahBest
</Text>
`;
